rest/route/annotations.go

```diff
@@ -297,10 +297,30 @@ func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request)
 		}
 	}
 
-	// set TaskExecution to the latest execution
-	if h.annotation.TaskExecution == nil {
-		h.annotation.TaskExecution = &t.Execution
+	if h.annotation.TaskExecution == 0 && taskExecutionsAsString == "" {
```

Comment:
it can be 0 because someone set it to the 0 execution. I think we do need a pointer still because a user can either set it in the body or the url and either is valid. cc @ablack12

Reply:
True, because as it stands if execution is omitted from the url it will be treated as 0.